transcode.py

```diff
@@ -1,92 +1,75 @@
-from dataclasses import dataclass
 import logging
 import os
 import time
+from base_util import Provenance, run_shell_command
 
-import base_util
-from config import data_base_dir
 
 logger = logging.getLogger(__name__)
 
 
-@dataclass
-class TranscodeOutput:
-    transcoded_file_path: str
-    provenance: dict
-    error: str = ""
-
-
-def try_transcode(input_path, asset_id, extension) -> TranscodeOutput:
+def try_transcode(
+    input_file: str,
+    asset_id: str,
+    extension: str,
+    output_path: str,
+) -> Provenance:
     logger.info(
-        f"Determining if transcode is required for input_path: {input_path} asset_id: ({asset_id}) extension: ({extension})"
+        f"Determining if transcode is required for input_path: {input_file} asset_id: ({asset_id}) extension: ({extension})"
     )
     start_time = time.time()
 
-    provenance = {
-        "activity_name": "Transcoding",
-        "activity_description": "Checks if input needs transcoding, then transcodes if so",
-        "processing_time_ms": -1,
-        "start_time_unix": start_time,
-        "parameters": [],
-        "software_version": "",
-        "input_data": input_path,
-        "output_data": "",
-        "steps": [],
-    }
-
-    # if it's already valid audio no transcode necessary
-    if _is_audio_file(extension):
-        logger.info("No transcode required, input is audio")
-        end_time = (time.time() - start_time) * 1000
-        provenance["processing_time_ms"] = end_time
-        provenance["output_data"] = input_path
-        provenance["steps"].append("No transcode required, input is audio")
-        return TranscodeOutput(input_path, provenance)
+    provenance = Provenance(
+        activity_name="Transcoding",
+        activity_description="Checks if input needs transcoding, then transcodes if so",
+        start_time_unix=start_time,
+        input_data=input_file,
+    )
+
+    output_file = os.path.join(output_path, f"{asset_id}.mp3")
+
+    # check if the input file needs transcoding (or is already transcoded)
+    if _is_audio_file(extension) or os.path.exists(output_file):
+        logger.info("Input is already audio or has been transcoded")
+        provenance.processing_time_ms = (time.time() - start_time) * 1000
+        provenance.output_data = output_path
+        provenance.steps.append("Input is already audio or has been transcoded")
```

Comment:
I don't particularly like using 'steps' for storing info/context, but that's something we can address later